### PR TITLE
Ignore coverage in multi-config generators

### LINKED ISSUE #267: Error during CMAKE for Lapack 3.8 with Visual Studio 2017
Hi,
Getting the following error during CMAKE with the following configurations:
- VS 2017 x64
- Intel Parallel Studio 2018 Update 3 installed
- Lapack 3.8.0

CMake is throwing an error here as below:
CMake Error at CMakeLists.txt:24 (string):
  string no output variable specified

CMake Error at CMakeLists.txt:25 (if):
  if given arguments:

    "STREQUAL" "COVERAGE"

  Unknown arguments specified

Tested with Lapack 3.7.1 and error is not there.

### PR DESCRIPTION
**Pull request by phcerdan**
`CMAKE_BUILD_TYPE` is only meaningful on single-configuration
generators (make, ninja). So the coverage logic was creating
problems on multi-config generators (Visual Studio, XCode).

Coverage is ignored in multi-generator generators.

Fix #267

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,12 +20,10 @@ SET(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   message(STATUS "Setting build type to 'Release' as none was specified.")
   set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
-  # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo" "Coverage")
 endif()
```
Comment (jcfr) [inline CMakeLists.txt]: Instead, consider adding this here:
```cmake
set(_msg "Checking if build type is 'Coverage'")
message(STATUS "${_msg}")

set(_is_coverage_build 0)
if(NOT CMAKE_CONFIGURATION_TYPES)
  set_property(CACHE CMAKE_BUILD_TYPE APPEND PROPERTY STRINGS "Coverage")

  string(TOLOWER ${CMAKE_BUILD_TYPE} _build_type_lc)
  if(${_build_type_lc} STREQUAL "coverage")
    set(_is_coverage_build 1)
  endif()
endif()
message(STATUS "${_msg}: ${_is_coverage_build}")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,12 +20,10 @@ SET(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
...
 
-string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
-if(${CMAKE_BUILD_TYPE_UPPER} STREQUAL "COVERAGE")
```
Comment (jcfr) [inline CMakeLists.txt]:
```cmake
if(_is_coverage_build)
```

File: SRC/CMakeLists.txt [hunk 3]
```diff
@@ -504,7 +504,7 @@ if(USE_XBLAS)
 endif()
 target_link_libraries(lapack PRIVATE ${BLAS_LIBRARIES})
 
-if (${CMAKE_BUILD_TYPE_UPPER} STREQUAL "COVERAGE")
```
Comment (jcfr) [inline SRC/CMakeLists.txt]:
```cmake
if(_is_coverage_build)
```

**Comment by juanprietob** [comment]
Great news!

**Comment by jcfr** [comment]
Thanks @julielangou

**Pull request merged**